cloudant-client/src/main/java/com/cloudant/client/api/Changes.java

```diff
@@ -1,309 +1,312 @@
+        this.databaseHelper.query("heartbeat", heartBeat);
+        return this;
+    }
+
+    /**
+     * Configure a timeout for the changes feed.
+     *
+     * @param timeout time in milliseconds to wait for data
+     * @return this Changes instance
+     */
+    public Changes timeout(long timeout) {
+        this.databaseHelper.query("timeout", timeout);
+        return this;
+    }
+
+    /**
+     * Specify a filter function to apply to the changes feed.
+     *
+     * @param filter name of the design document filter function e.g {@code
+     * "designDoc/filterFunction"}
+     * @return this Changes instance
+     */
+    public Changes filter(String filter) {
+        this.databaseHelper.query("filter", filter);
+        return this;
+    }
+
+    /**
+     * @param includeDocs whether to include document content in the returned rows
+     * @return this Changes instance
+     */
+    public Changes includeDocs(boolean includeDocs) {
+        this.databaseHelper.query("include_docs", includeDocs);
+        return this;
+    }
+
+    /**
+     * Configures how many changes are returned "main_only" for the winning revision only or
+     * "all_docs" to also include leaf revisions.
+     *
+     * @param style {@code "main_only"} or {@code "all_docs"}
+     * @return this Changes instance
+     */
+    public Changes style(String style) {
+        this.databaseHelper.query("style", style);
+        return this;
+    }
+
+    /**
+     * @param descending {@code true} to return changes in descending order
+     * @return this Changes instance
+     * @since 2.5.0
+     */
+    public Changes descending(boolean descending) {
+        this.databaseHelper.query("descending", descending);
+        return this;
+    }
+
+    /**
+     * Add a custom query parameter to the _changes request. Useful for specifying extra parameters
+     * to a filter function for example.
+     *
+     * @param name  the name of the query parameter
+     * @param value the value of the query parameter
+     * @return this Changes instance
+     * @since 2.5.0
+     */
+    public Changes parameter(String name, String value) {
+        this.databaseHelper.query(name, value);
+        return this;
+    }
+
+    // Helper
+
+    /**
+     * Reads and sets the next feed in the stream.
+     */
+    private boolean readNextRow() {
+        while (!stop) {
+            String row = getLineWrapped();
+            // end of stream - null indicates end of stream before we see last_seq which shouldn't
+            // be possible but we should handle it
+            if (row == null || row.startsWith("{\"last_seq\":")) {
+                terminate();
+                return false;
+            } else if (row.isEmpty()) {
+                // heartbeat
+                continue;
+            }
+            setNextRow(gson.fromJson(row, ChangesResult.Row.class));
+            return true;
+        }
+        // we were stopped, end of changes feed
+        return false;
```

Comment:
do we need to call `terminate();` here too?

Reply:
Good point, done in 653c4cb